code/frameworks/nextjs/README.md

```diff
@@ -0,0 +1,622 @@
+  - [Css/Sass/Scss Modules](#csssassscss-modules)
+  - [Styled JSX](#styled-jsx)
+  - [Postcss](#postcss)
+  - [Absolute Imports](#absolute-imports)
+  - [Runtime Config](#runtime-config)
+  - [Custom Webpack Config](#custom-webpack-config)
+  - [Typescript](#typescript)
+  - [next.config.js](#nextconfigjs)
+    - [ESM](#esm)
+  - [Notes for Yarn v2 and v3 users](#notes-for-yarn-v2-and-v3-users)
+  - [FAQ](#faq)
+    - [Stories for pages](#stories-for-pages)
+    - [Statically imported images won't load](#statically-imported-images-wont-load)
+    - [This framework breaks when the .mjs extension for the next config is used](#this-framework-breaks-when-the-mjs-extension-for-the-next-config-is-used)
+    - [Module not found: Error: Can't resolve [package name]](#module-not-found-error-cant-resolve-package-name)
+- [Acknowledgements](#acknowledgements)
+
+## Supported Features
+
+👉 [Next.js's Image Component](#nextjss-image-component)
+
+👉 [Next.js Routing](#nextjs-routing)
+
+👉 [Sass/Scss](#sassscss)
+
+👉 [Css/Sass/Scss Modules](#csssassscss-modules)
+
+👉 [Styled JSX](#styled-jsx)
+
+👉 [Postcss](#postcss)
+
+👉 [Absolute Imports](#absolute-imports)
+
+👉 [Runtime Config](#runtime-config)
+
+👉 [Custom Webpack Config](#custom-webpack-config)
+
+👉 [Typescript](#typescript) (already supported out of the box by Storybook)
+
+## Requirements
+
+- [Next.js](https://nextjs.org/) >= 9.x
+- [Storybook](https://storybook.js.org/) >= 7.x
+- Your Next.js config file uses the `.js` extension and not the `.mjs` extension (i.e. `next.config.js` not `next.config.mjs`)
+  - See [next.config.js](#nextconfigjs) for more details
+
+## Getting Started
+
+### In a project without Storybook
+
+Follow the prompts after running this command in your Next.js project's root directory:
+
+```bash
+npx storybook init
+```
+
+[More on getting started with Storybook](https://storybook.js.org/docs/react/get-started/introduction)
+
+### In a project with Storybook
+
+Update your `main.js` to look something like this:
+
+Install the framework:
+
+```bash
+yarn install @storybook/nextjs
+```
+
+```js
+// .storybook/main.js
+module.exports = {
+  framework: {
+    name: '@storybook/nextjs',
+    options: {};
+  }
+}
+```
+
+## Documentation
+
+### Options
+
+You can be pass an options object for addional configuration if needed.
+
+For example:
+
+```js
+// .storybook/main.js
+const path = require('path');
+
+module.exports = {
+  // other config ommited for brevity
+  framework: {
+    name: '@storybook/nextjs',
+    options: {
+      nextConfigPath: path.resolve(__dirname, '../next.config.js'),
+    },
+  },
+  // ...
+};
+```
+
+- `nextConfigPath`: The absolute path to the `next.config.js`
+
+### Next.js's Image Component
+
+[next/image](https://nextjs.org/docs/api-reference/next/image) is [notoriously difficult](https://github.com/vercel/next.js/issues/18393) to get working with Storybook. This framework allows you to use Next.js's `Image` component with no configuration!
```

Comment:
also heads up: before releasing this framework, consider including support for `next/future/image`. I've been getting a bunch of github issues to support it. The workaround I do in my addon for the regular image should be a cookie cutter solution for the future image. (There's also an open issue on my addon's repo with some other workarounds people made)

Reply:
For reference: https://github.com/RyanClementsHax/storybook-addon-next/issues/99

Reply:
I suspect the main difficulty here is going to be conditionally including whatever code ends up being used from the above reference (e.g. we can't `require('next/future/image')` if the project is using a version of `next` that doesn't include that package).

Additionally, it's not likely imo, but we may need to update the selector [mentioned here](https://github.com/storybookjs/storybook/pull/19382#discussion_r989321448) to work with both `next/image` (which includes wrapping markup around the `img`)  _and_ `next/future/image` (which does not).